xds/src/main/java/io/grpc/xds/XdsClusterResource.java

```diff
@@ -536,8 +525,6 @@ static void validateCommonTlsContext(
   private static String getIdentityCertInstanceName(CommonTlsContext commonTlsContext) {
     if (commonTlsContext.hasTlsCertificateProviderInstance()) {
       return commonTlsContext.getTlsCertificateProviderInstance().getInstanceName();
-    } else if (commonTlsContext.hasTlsCertificateCertificateProviderInstance()) {
```

Comment:
Ugh. Apparently these might actually be needed. TODO: investigate

Reply:
It turns out it was used for a while, but no longer. So it should still be safe to remove.